services/s3management/accesskeys.go

```diff
@@ -0,0 +1,165 @@
+	apiResponse, err := c.client.AccesskeysApi.AccesskeysDelete(ctx, accessKeyId).Execute()
+	apiResponse.LogInfo()
+
+	if err == nil {
+		err = backoff.Retry(func() error {
+			return c.accessKeyDeletedCheck(ctx, accessKeyId)
+		}, backoff.NewExponentialBackOff(backoff.WithMaxElapsedTime(timeout)))
+		if err != nil {
+			return apiResponse, fmt.Errorf("failed to wait for accessKey available: %w", err)
+		}
+	}
+
+	return apiResponse, err
+}
+
+func SetAccessKeyPropertiesToPlan(plan *AccesskeyResourceModel, accessKey s3management.AccessKey) {
+
+	if accessKey.Properties != nil {
+		if accessKey.Properties.AccessKey != nil {
+			plan.AccessKey = basetypes.NewStringPointerValue(accessKey.Properties.AccessKey)
+		}
+		if accessKey.Properties.CanonicalUserId != nil {
+			plan.CanonicalUserId = basetypes.NewStringPointerValue(accessKey.Properties.CanonicalUserId)
+		}
+		if accessKey.Properties.ContractUserId != nil {
+			plan.ContractUserId = basetypes.NewStringPointerValue(accessKey.Properties.ContractUserId)
+		}
+		if accessKey.Properties.Description != nil {
+			plan.Description = basetypes.NewStringPointerValue(accessKey.Properties.Description)
+		}
+		if accessKey.Properties.SecretKey != nil {
+			plan.SecretKey = basetypes.NewStringPointerValue(accessKey.Properties.SecretKey)
+		}
+	}
+	if accessKey.Id != nil {
+		plan.ID = basetypes.NewStringPointerValue(accessKey.Id)
+	}
+}
+
+func SetAccessKeyPropertiesToDataSourcePlan(plan *AccessKeyDataSourceModel, accessKey s3management.AccessKey) {
+
+	if accessKey.Properties != nil {
+		if accessKey.Properties.AccessKey != nil {
+			plan.AccessKey = basetypes.NewStringPointerValue(accessKey.Properties.AccessKey)
+		}
+		if accessKey.Properties.CanonicalUserId != nil {
+			plan.CanonicalUserId = basetypes.NewStringPointerValue(accessKey.Properties.CanonicalUserId)
+		}
+		if accessKey.Properties.ContractUserId != nil {
+			plan.ContractUserId = basetypes.NewStringPointerValue(accessKey.Properties.ContractUserId)
+		}
+		if accessKey.Properties.Description != nil {
+			plan.Description = basetypes.NewStringPointerValue(accessKey.Properties.Description)
+		}
+	}
+	if accessKey.Id != nil {
+		plan.ID = basetypes.NewStringPointerValue(accessKey.Id)
+	}
+}
+
+func (c *Client) accessKeyDeletedCheck(ctx context.Context, id string) error {
+	_, apiResponse, err := c.GetAccessKey(ctx, id)
+	if apiResponse.HttpNotFound() {
+		return nil
+	}
+
+	if err != nil {
+		return backoff.Permanent(fmt.Errorf("failed to check if accessKey exists: %w", err))
+	}
+
+	return fmt.Errorf("accessKey still exists")
+}
+
+func (c *Client) accessKeyAvailableCheck(ctx context.Context, id string) error {
+	accessKey, apiResponse, err := c.GetAccessKey(ctx, id)
+	if apiResponse.HttpNotFound() {
+		return fmt.Errorf("accessKey not found")
+	}
+
+	if err != nil {
+		return backoff.Permanent(fmt.Errorf("failed to check if accessKey exists: %w", err))
+	}
+
+	if *accessKey.Metadata.Status != "AVAILABLE" {
```

Comment:
do we have an enum here that we can use?

Reply:
unfortunately there is not